crates/transaction-pool/src/traits.rs

```diff
@@ -272,6 +275,121 @@ pub trait PoolTransaction: fmt::Debug + Send + Sync + FromRecoveredTransaction {
     fn size(&self) -> usize;
 }
 
+// #[main_codec]
+#[derive(Debug, Clone, PartialEq, Eq, Hash, Default)]
+pub(crate) struct PooledTransaction {
+    /// EcRecovered transaction info
+    pub(crate) transaction: TransactionSignedEcRecovered,
+
+    /// Hash of the transaction.
+    pub(crate) hash: TxHash,
```

Comment:
I also realized, this is already available at `TransactionSignedEcRecovered.transaction.hash`